app/actions/remote/post.ts

```diff
@@ -383,33 +382,21 @@ export const fetchPostsForUnreadChannels = async (
             // If team index is the same, sort by last_viewed_at in descending order
             return b.last_viewed_at - a.last_viewed_at;
         }).
-        map((member) => member.channel_id);
-
-    const postsForChannel: PostsForChannel[] = [];
-    const posts: Post[] = [];
-
-    // process 10 unread channels at a time
-    const chunks = chunk(sortedUnreadchannelIds, 10);
-    for await (const channelIds of chunks) {
-        const promises = [];
-        for (const channelId of channelIds) {
-            promises.push(fetchPostsForChannel(serverUrl, channelId, fetchOnly, true, groupLabel));
-        }
-
-        const results = await Promise.all(promises);
-        postsForChannel.push(...results);
-        for (const r of results) {
-            if (r.posts?.length) {
-                posts.push(...r.posts);
+        reduce((acc, member) => {
+            const channel = channelsMap.get(member.channel_id);
+            const teamId = channel!.team_id || '';
+            if (!acc[teamId]) {
+                acc[teamId] = [];
             }
-        }
-    }
+            acc[teamId].push(channel!.id);
 
-    if (posts.length) {
-        fetchPostAuthors(serverUrl, posts, false, groupLabel);
-    }
+            return acc;
+        }, {} as Record<string, string[]>);
 
-    return postsForChannel;
```

Comment:
I removed the need to return postsForChannel since the caller is not using it for anything.  If we're cumulating all channels from all teams, it might take up memory for no reason.